web/src/main/webapp/dataViews/schemaView.ts

```diff
@@ -177,6 +177,12 @@ export class SchemaView extends TSViewBase {
             help: "Plot the data in the selected columns as a histogram.  Applies to one or two columns only. " +
             "The data cannot be of type String.",
         }, selectedCount >= 1 && selectedCount <= 2);
+        this.contextMenu.addItem({
+            text: "Private histogram",
+            action: () => this.histogramSelected(),
+            help: "Plot the data in the selected columns as a private histogram. " +
+                "Applies to one numeric column only.",
+        }, selectedCount === 1);
```

Comment:
You can also probably check here that the column is convertible to numeric in the enable condition.